### DISCUSSION TITLE
How to just track a part of particle.

### DISCUSSION THREAD

**Discussion by jiecaix** [BLAST-WarpX/warpx#4756] [Q&A] [answered]
I want track some particles, and the "random_fraction" flag same to just random sample at each timestep, so that i can't just track one particle from start to end (most time it may be not be random sampled). 

By the way, is there anyway to set a tracer flag for electrons so that it didn't contribute the charge and density for simulation but just for track?

**Comment by GivenChen** [comment, marked as answer]
yeah for the second question, I think when you inject species, you have the option to set "do not deposit", so the species won't contribute to the charge and current..

**Comment by ax3l** [comment]
Using a separate species as @GivenChen suggests is the right approach for now.

Alternatively, we can in the future also pass the particle id into
```
<diag_name>.<species_name>.plot_filter_function(t,x,y,z,ux,uy,uz)
```
but that needs to solve some details about the id type (so far, all filter arguments are floating point numbers).
